Hashtable.c

```diff
@@ -88,32 +90,91 @@ size_t Hashtable_count(const Hashtable* this) {
-   34359738337, 68719476731, 137438953447
+#define MIN_TABLE_SIZE 11
+
+/* Primes borrowed from gnulib/lib/gl_anyhash_primes.h.
+
+   Array of primes, approximately in steps of factor 1.2.
+   This table was computed by executing the Common Lisp expression
+     (dotimes (i 244) (format t "nextprime(~D)~%" (ceiling (expt 1.2d0 i))))
+   and feeding the result to PARI/gp. */
+static const size_t primes[] = {
+   MIN_TABLE_SIZE, 13, 17, 19, 23, 29, 37, 41, 47, 59, 67, 83, 97, 127, 139,
+   167, 199, 239, 293, 347, 419, 499, 593, 709, 853, 1021, 1229, 1471, 1777,
+   2129, 2543, 3049, 3659, 4391, 5273, 6323, 7589, 9103, 10937, 13109, 15727,
+   18899, 22651, 27179, 32609, 39133, 46957, 56359, 67619, 81157, 97369,
+   116849, 140221, 168253, 201907, 242309, 290761, 348889, 418667, 502409,
+   602887, 723467, 868151, 1041779, 1250141, 1500181, 1800191, 2160233,
+   2592277, 3110741, 3732887, 4479463, 5375371, 6450413, 7740517, 9288589,
+   11146307, 13375573, 16050689, 19260817, 23112977, 27735583, 33282701,
+   39939233, 47927081, 57512503, 69014987, 82818011, 99381577, 119257891,
+   143109469, 171731387, 206077643, 247293161, 296751781, 356102141, 427322587,
+   512787097, 615344489, 738413383, 886096061, 1063315271, 1275978331,
+   1531174013, 1837408799, 2204890543UL, 2645868653UL, 3175042391UL,
+   3810050851UL,
+   /* on 32-bit make sure we do not return primes not fitting in size_t */
+#if SIZE_MAX > 4294967295ULL
+   4572061027ULL, 5486473229ULL, 6583767889ULL, 7900521449ULL, 9480625733ULL,
+   /* Largest possible size should be 13652101063ULL == GROWTH_RATE((UINT32_MAX/3)*4)
+      we include some larger values in case the above math is wrong */
+   11376750877ULL, 13652101063ULL, 16382521261ULL, 19659025513ULL, 23590830631ULL,
+#endif
 };
 
 static size_t nextPrime(size_t n) {
-   /* on 32-bit make sure we do not return primes not fitting in size_t */
-   for (size_t i = 0; i < ARRAYSIZE(OEISprimes) && OEISprimes[i] < SIZE_MAX; i++) {
-      if (n <= OEISprimes[i])
-         return OEISprimes[i];
+   for (size_t i = 0; i < ARRAYSIZE(primes); i++) {
+      if (n < primes[i]) {
+         return primes[i];
+      }
    }
 
    CRT_fatalError("Hashtable: no prime found");
 }
 
-Hashtable* Hashtable_new(size_t size, bool owner) {
-   Hashtable* this;
+/* USABLE_FRACTION is the maximum hash map load.
+ * Currently set to 2/3 capacity.
+ *
+ * Testing indicates that the median and average probe length
+ * increases significantly after 2/3 of the hash map capacity.
+ *
+ * load = {size,capacity} / items
+ *
+ * | load | probe max | probe avg | probe median |
+ * | ---- | --------- | --------- | ------------ |
+ * | 0.60 |     90.58 |     20.19 |         0.65 |
+ * | 0.65 |    167.00 |     37.07 |         1.58 |
+ * | 0.70 |    230.54 |     61.40 |        15.54 |
+ * | 0.75 |    287.00 |     85.23 |        26.15 |
+ * | 0.80 |    287.00 |     94.71 |        55.93 |
+ */
+#define USABLE_FRACTION(n) (((n) << 1)/3)
```

Comment:
Prefer normal multiplication.
```suggestion
#define USABLE_FRACTION(n) (((n) * 2)/3)
```

Reply:
Another point that should be integrated into this patch series …